packages/components/src/RecurringSelect/components/WeeklySelect/WeeklySelect.css

```diff
@@ -0,0 +1,75 @@
+/* 
+  faking a border with a background on the parent element, gaps & padding
+  to prevent double borders on overlapping boxes
+*/
+.container {
+  display: grid;
+  width: 100%;
+  padding: var(--space-smaller);
+  border: var(--border-base) solid var(--color-border);
+  border-radius: var(--radius-large);
+  grid-template-columns: repeat(7, 1fr);
+  row-gap: var(--space-smaller);
+  column-gap: var(--space-smaller);
+}
+
+.checkboxWrapper {
+  position: relative;
+}
+
+/* kept present but hidden for accessibility */
+.container input[type="checkbox"] {
```

Comment:
General comment/question (and it's fine IMO if the answer is "we can't do it"), but could we extract up some of this CSS? We're writing the same styling in a few places throughout the larger component.

Reply:
I'm getting so sad from these suggestions that I fully agree with, but run into pretty big blockers when trying to implement :(

so, I can indeed move the shared styles up into their own file that might look like this

```
RecurringSelect
-- DateCellCheckbox.css
-- components
---- DayOfMonthSelect
---- WeeklySelect
```

I can also import those top level styles into those `components`

as is tradition, things fall apart when I try to actually use those imported styles

in order to avoid polluting the global space with a rule like `input[type="checkbox"] { display: none }` we prefixed it with the scoped component class of `container` that you can see in the line this comment/thread was started off of

what the problem is here, is that when the styles are up in `DateCellCheckbox.css` `.container` is not the same `.container` as the one down in the component where we use the styles

this leaves us with 2 options:

import the `DateCellCheckbox.css` into the top level RecurringSelect component despite it having no actual interest in checkboxes at that level. then add a div that surrounds the element that will be swapped out with the various selects (weekly, monthly etc.). on this div, apply the `DateCellCheckbox.container` and now all the rules will be correctly referenced and scoped. once again we have the issue of a useless css class ruleset, because the linter doesn't like empty rules, but we don't actually have any css to apply here. all we want is the scoping and reference to the correct container :| (this is turning into a pattern)

OR perhaps add a new class to identify the `input[type="checkbox"]` which allows us to not need to scope it anymore

I went with approach 1 (it works!) because I was troubleshooting and trying to understand the exact mechanics of the issue, but I could swap to the second if we really want!

Reply:
jk went with option 2 and it's much much better